### ISSUE TITLE
Product option choice considered a soft navigation

### ISSUE THREAD

**Issue by yoavweiss** [WICG/soft-navigations#7]
[crbug/1371927](https://bugs.chromium.org/p/chromium/issues/detail?id=1371927) outlines a few examples where the heuristic's result is arguably a false positive: product options where the user clicks to e.g. change the item's color result in a URL change and a DOM modification, and hence are considered a soft navigation.

More research is needed to see if this can be avoided.

**Comment by yoavweiss** [comment]
After looking at the examples again, I'm less confident that it's a false positive:
* In many cases, changing an option reloads the hero image, showing the new product option (sometimes on a different model)
* It triggers many changes on the page such as: different thumbnails, price changes, different suggested items, etc

So it seems like it may be reasonable to consider this a soft navigation to a similar (yet different) product page.

^^ @anniesullie

**Comment by tunetheweb** [comment]
I agree. To me there's a user interaction, a state change, a UI change, and a URL change. That to me seems like it _should_ be a soft nav.

The only concern is if there is not a navigation history. On the Walmart example, for example, clicking back does not take me to the previous option selection.

So, rather than a URL update, should we insist on a URL `pushState` and exclude `replaceState`? Perhaps that should be a separate issue though...

**Comment by anniesullie** [comment]
> So it seems like it may be reasonable to consider this a soft navigation to a similar (yet different) product page

Would love to collect a list of issues like this where it's a bit subjective whether it's a navigation and hear working group feedback!

**Comment by yoavweiss** [comment]
The reliance on contentful paints should potentially help here.
If we see a bunch of cases where the change is "too small" to be considered a navigation, we could add a limit based on the %age of viewport changed. However, experiments with that approach did not prove particularly useful.

**Comment by mmocny** [comment]
I think that if a hard-nav website used cross-document links to move between views of a product, this would not be considered a back practice or bad ux or "not a navigation".

I think the expectation of rapid feedback for such interactions has led many developers to use Javascript to dynamically update content in place, and its good accessibility to also update URL for link-ability and history stack.

You can also get rapid cross document navigations-- so I think its fine and expected to report for such use cases.

**Comment by tunetheweb** [comment]
It appears these are no longer registering as soft navs (maybe because no History state update?)

So are we still OK closing this? Or have we now the opposite problem (they are NOT registering as soft navs when they should be)? Or is the last of history state the decider?

**Comment by mmocny** [comment]
As per the original issue description, if an interaction updates both URL and page content, I think it *should* be a soft-nav regardless if it is just a simple form that updates search criteria or image.

If the original URLs actually do these things but don't actually report soft-nav entries, that might be a chromium bug and not a spec issue.

---

I looked at the first ~6 urls in the chromium issue and found:
- 2 sites no longer appear to have the pattern (either dont soft-nav or I cannot find options buttons that soft-nav)
- 2 sites still have the pattern but dont report soft-navs
- 2 sites still have the pattern and do report soft-navs

I think we can trace these (walmart, and mercadolibre) to find what is causing the issue.
